llama_index/agent/openai_agent.py

```diff
@@ -136,6 +142,10 @@ def all_messages(self) -> List[ChatMessage]:
     def latest_function_call(self) -> Optional[dict]:
         return self.memory.get_all()[-1].additional_kwargs.get("function_call", None)
 
+    @property
+    def latest_tool_calls(self) -> Optional[List[dict]]:
+        return self.memory.get_all()[-1].additional_kwargs.get("tool_calls", None)
```

Comment:
do we put the raw openai object or the dict into `tool_calls`? it feels cleaner to me to work with typed objects than dictionary lookups e.g. see `for tool_call in tool_calls` in `openai_assistant_agent.py`

Reply:
Yeah i'm currently convert them to dict after receiving them. But yeah I agree with you that its cleaner to use the typed objects over dictionary lookups. Question re: those types, I don't believe we actually bridge them over to our library. should we?

Marking a TODO here for now to change this to typed objects over dicts.